File: 프로그래머스/lv2/김원래/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

stack<int> st;

bool solution(string s)
{
    bool answer = true;

    for(int cnt = 0 ; cnt < s.size() ; ++cnt)
    {
        if(s[cnt] == '(')
        {
            st.push(1);
        }
        else
        {
            if(st.empty())
            {
                answer = false;
                return answer;
            }
            st.pop();
        }
    }
    
    if(!st.empty()) answer = false;
    else answer = true;

    return answer;
}

File: 프로그래머스/lv2/이재표/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character>stack=new Stack<>();
        for(int idx1=0;idx1<s.length();idx1++){ //주어진 s를 돌기위한 반복문
            if(stack.size()==0){
                if(s.charAt(idx1)==')'){
                    return false;
                }
                stack.push(s.charAt(idx1));
                continue;
            }
            char peek=stack.peek();
            if(peek=='('){
                if(s.charAt(idx1)==')'){
                    stack.pop();
                }else{
                    stack.push(s.charAt(idx1));
                }
            }
        }
        if(stack.size()>0){
            return false;
        }
        return true;
    }
}
